src/cockpit/channels/filesystem.py

```diff
@@ -158,6 +160,64 @@ def do_yield_data(self, options: JsonObject) -> Generator[bytes, None, JsonObjec
             raise ChannelError('internal-error', message=str(exc)) from exc
 
 
+class FSReplaceAttrs:
+    supported_attrs = ['uid', 'gid', 'owner', 'group']
+
+    def __init__(self, value: JsonObject) -> None:
+        # Any unknown keys throw an error
+        unsupported_attrs = list(value.keys() - self.supported_attrs)
+        if unsupported_attrs:
+            raise ChannelError('protocol-error',
+                               message=f'"attrs" contains unsupported key(s) {unsupported_attrs}') from None
+
+        self.uid = get_int(value, 'uid', None)
+        self.gid = get_int(value, 'gid', None)
+
+        if self.uid is None and self.gid is not None:
+            raise ChannelError('protocol-error', message='cannot provide a gid without an uid')
+
+        try:
+            owner = get_str_or_int(value, 'owner', None)
+        except JsonError:
+            raise ChannelError('protocol-error', message='"owner" must be an integer or string') from None
+
+        try:
+            group = get_str_or_int(value, 'group', None)
+        except JsonError:
+            raise ChannelError('protocol-error', message='"group" must be an integer or string') from None
+
+        if owner is None and group is not None:
+            raise ChannelError('protocol-error', message='cannot provide a group without an owner')
+
+        if owner is not None:
+            if isinstance(owner, str):
+                try:
+                    uid = pwd.getpwnam(owner).pw_uid
+                except KeyError:
+                    raise ChannelError('internal-error', message=f'uid not found for {owner}') from None
+
+                if self.uid is not None and self.uid != uid:
+                    raise ChannelError('protocol-error', message='"owner" uid does not match passed "uid"') from None
```

Comment:
I find this rather weird -- would it be enough to just support "owner" and "group" and leave it at that? We can't support all the `fsinfo` attributes anyway (like type or size).